src/main/java/se/kth/dd2480/grp25/ci/BuildJob.java

```diff
@@ -0,0 +1,94 @@
+package se.kth.dd2480.grp25.ci;
+
+import java.io.File;
+import java.nio.file.NotDirectoryException;
+import java.util.Optional;
+import org.gradle.tooling.*;
+
+/** A job that prints every event to stdout. */
+public class BuildJob implements Runnable {
+  /**
+   * This functions decides if it wants to accept an event and offer a {@link PrintJob}.
+   *
+   * <p>This function confirms to the {@link JobAcceptor} interface.
+   *
+   * @param event the event offered to this function to accept or decline
+   * @return an print job represented by a {@link Runnable} if accepted
+   */
+  public static Optional<Runnable> offer(Event event) {
+    // 1. Decide if we want to handle this event
+    if (event.getType() == Event.EventType.BUILD) {
+      // 2. If we do, return a job handler
+      return Optional.of(new BuildJob(event));
+    } else {
+      // 3. If we don't, return nothing
+      return Optional.empty();
+    }
+  }
+
+  private Event event;
+  private String path;
+
+  public BuildJob(Event event) {
+    this.event = event;
+    this.path = "./" + event.getId();
+  }
+
+  /**
+   * Opens a connection to a project through given path and launches a build via help function
+   * 'launch()'
+   */
+  @Override
+  public void run() {
+
+    try {
+      File project = (new File(path));
+
+      if (!project.exists()) {
+        event.setStatusCode(Event.StatusCode.FAIL);
+        throw new NotDirectoryException("project directory not found at: " + path);
```

Comment:
I don't think it is necessary to throw an exception here since you set the StatusCode to FAIL. It would be good to set the message too, event.setMessage("some error message");